core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala

```diff
@@ -186,34 +186,31 @@ class KafkaConfigTest {
 
     // listeners with duplicate port
     props.put(KafkaConfig.ListenersProp, "PLAINTEXT://localhost:9091,SSL://localhost:9091")
-    var caught = assertThrows(classOf[IllegalArgumentException], () => KafkaConfig.fromProps(props))
-    assertTrue(caught.getMessage.contains("Each listener must have a different port"))
+    assertBadConfigContainingMessage(props, "Each listener must have a different port")
 
     // listeners with duplicate name
     props.put(KafkaConfig.ListenersProp, "PLAINTEXT://localhost:9091,PLAINTEXT://localhost:9092")
-    caught = assertThrows(classOf[IllegalArgumentException], () => KafkaConfig.fromProps(props))
-    assertTrue(caught.getMessage.contains("Each listener must have a different name"))
+    assertBadConfigContainingMessage(props, "Each listener must have a different name")
 
     // advertised listeners can have duplicate ports
     props.put(KafkaConfig.ListenerSecurityProtocolMapProp, "HOST:SASL_SSL,LB:SASL_SSL")
     props.put(KafkaConfig.InterBrokerListenerNameProp, "HOST")
     props.put(KafkaConfig.ListenersProp, "HOST://localhost:9091,LB://localhost:9092")
     props.put(KafkaConfig.AdvertisedListenersProp, "HOST://localhost:9091,LB://localhost:9091")
-    assertTrue(isValidKafkaConfig(props))
+    KafkaConfig.fromProps(props)
```

Comment:
Simply trying to create the config is better because if the config is invalid we actually see the exception that caused it, which is much more helpful than simply seeing that `true` was expected and we got `false` instead.  I made this change in many places below as well.